kubetest/azure.go

```diff
@@ -0,0 +1,311 @@
+		return fmt.Errorf("Failed to generate ARM templates: %v", err)
+	}
+	return nil
+}
+
+func (c *Cluster) loadARMTemplates() error {
+	var err error
+	template, err := ioutil.ReadFile(path.Join(c.outputDir, "azuredeploy.json"))
+	if err != nil {
+		return fmt.Errorf("Error reading ARM template file: %v", err)
+	}
+	err = json.Unmarshal([]byte(template), &c.templateJSON)
+	if err != nil {
+		return fmt.Errorf("Error unmarshall template %v", err.Error())
+	}
+
+	parameters, err := ioutil.ReadFile(path.Join(c.outputDir, "azuredeploy.parameters.json"))
+	if err != nil {
+		return fmt.Errorf("Error reading ARM parameters file: %v", err)
+	}
+	err = json.Unmarshal([]byte(parameters), &c.parametersJSON)
+	if err != nil {
+		return fmt.Errorf("Error unmarshall parameters %v", err.Error())
+	}
+	return nil
+}
+
+func (c *Cluster) getARMClient() error {
+	env, err := azure.EnvironmentFromName("AzurePublicCloud")
+	var client *AzureClient
+	if client, err = getAzureClient(env,
+		c.subscriptionId,
+		c.clientId,
+		c.clientSecret); err != nil {
+		return fmt.Errorf("Error trying to get Azure Client: %v", err)
+	}
+	c.azureClient = client
+	return nil
+}
+
+func (c *Cluster) createCluster() error {
+	var err error
+	err = c.getARMClient()
+	if err != nil {
+		return fmt.Errorf("Failed to generate ARM client: %v", err)
+	}
+	kubecfgDir, _ := ioutil.ReadDir(path.Join(c.outputDir, "kubeconfig"))
+	kubecfg := path.Join(c.outputDir, "kubeconfig", kubecfgDir[0].Name())
+	fmt.Printf("Setting kubeconfig env variable: kubeconfig path: %v", kubecfg)
+	os.Setenv("KUBECONFIG", kubecfg)
+	_, err = c.azureClient.EnsureResourceGroup(c.resourceGroup, c.location, nil)
+	if err != nil {
+		return fmt.Errorf("Could not ensure resource group: %v", err)
+	}
+
+	if res, err := c.azureClient.DeployTemplate(
+		c.resourceGroup,
+		c.name,
+		c.templateJSON,
+		c.parametersJSON,
+		nil); err != nil {
+		if res != nil && res.Response.Response != nil && res.Body != nil {
+			defer res.Body.Close()
+			body, _ := ioutil.ReadAll(res.Body)
+			fmt.Printf(string(body))
+		}
+		fmt.Printf("Cannot deploy: %v", err)
+	}
+	return nil
+
+}
+
+func (c Cluster) Up() error {
+	var err error
+	err = c.generateTemplate()
+	if err != nil {
+		return fmt.Errorf("Failed to generat apiModel: %v", err)
+	}
+	err = c.getAcsEngine(1)
+	if err != nil {
+		return fmt.Errorf("Failed to get ACS Engine binary: %v", err)
+	}
+	err = c.generateARMTemplates()
+	if err != nil {
+		return fmt.Errorf("Failed to generate ARM templates: %v", err)
+	}
+	err = c.loadARMTemplates()
+	if err != nil {
+		return fmt.Errorf("Error loading ARM templates: %v", err)
+	}
+	err = c.createCluster()
+	if err != nil {
+		return fmt.Errorf("Error creating cluster: %v", err)
+	}
+
+	return nil
+}
+
+func (c Cluster) Down() error {
+	return nil
+}
+
+func (c Cluster) DumpClusterLogs(localPath, gcsPath string) error {
+	return nil
+}
+
+func (c Cluster) GetClusterCreated(clusterName string) (time.Time, error) {
+	return time.Now(), nil
```

Comment:
Just 'return time.Time{}, errors.New("not implemented")' as others. time.Now is not the time cluster created.